### PR TITLE
SAA-687 - Made appointment descriptions null if blank string

### PR DESCRIPTION
**Pull request by sioldham**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt [hunk 1]
```diff
@@ -156,7 +156,7 @@ class AppointmentService(
       startTime = startTime!!,
       endTime = endTime,
       comment = comment,
-      appointmentDescription = appointmentDescription,
+      appointmentDescription = if (appointmentDescription?.isBlank() == true) null else appointmentDescription,
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt:159]: Some more Kotlin magic only if you want to ...

`appointmentDescription = appointmentDescription?.takeUnless(String::isBlank)`

**Review by matt-crowson-moj — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt (hunk 1 above)
Comment (sioldham) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt:159]: Nice! Kotlin wizardry! 🧙‍♂️

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Review by iterative-it — APPROVED** [review]
LGTM

**Pull request merged**